meson.build

```diff
@@ -43,18 +44,27 @@ kanshi_deps = [
 	client_protos,
 ]
 
+kanshi_args = ['-DKANSHI_VERSION="@0@"'.format(meson.project_version())]
+
 kanshi_srcs = [
 	'event-loop.c',
 	'main.c',
 	'parser.c',
 	'ipc-addr.c',
 ]
 
+if varlink.found()
+	kanshi_deps += varlink
+	kanshi_args += '-DKANSHI_HAS_VARLINK'
```

Comment:
I'd prefer to use `add_project_arguments` instead of `kanshi_args` to ensure the defines are propagated to the whole project, even if we change the build system in the future.

I'd also prefer to use `-DKANSHI_HAS_VARLINK=1` to indicate that libvarlink is available, and `DKANSHI_HAS_VARLINK=0` otherwise, then test with `#if` instead of `#ifdef`. Together with `-Wundef` this makes sure the define isn't lost.

Reply:
Heh, I had exactly that issue when I was testing the PR :p 

One file had struct with varlink members, the other didn't. Using global args sounds much better indeed.